pkg/gofr/datasource/dgraph/dgraph.go

```diff
@@ -283,9 +269,29 @@ func (d *Client) HealthCheck(ctx context.Context) (any, error) {
 	return "UP", nil
 }
 
-func (d *Client) sendOperationStats(ctx context.Context, query *QueryLog, metricName string) {
-	query.PrettyPrint(d.logger)
-	d.metrics.RecordHistogram(ctx, metricName, float64(query.Duration))
+func (d *Client) addTrace(ctx context.Context, method string) (context.Context, trace.Span) {
+	if d.tracer != nil {
+		tracedCtx, span := d.tracer.Start(ctx, fmt.Sprintf("dgraph-%v", method))
+
+		return tracedCtx, span
+	}
+
+	return ctx, nil
+}
```

Comment:
The code would be cleaner


```suggestion
	if d.tracer == nil {
		return ctx, nil
	}
	tracedCtx, span := d.tracer.Start(ctx, fmt.Sprintf("dgraph-%v", method))
	return tracedCtx, span
}
```